### ISSUE TITLE
segfault in selection_changed_at_idle_callback

### ISSUE THREAD

**Issue by brianjmurrell** [mate-desktop/caja#289]
Per the downstream bug report https://bugzilla.redhat.com/show_bug.cgi?id=1066147 there's a segfault in selection_changed_at_idle_callback with the following stack trace:

```
Thread 1 (Thread 0x7f46445e2a40 (LWP 1323)):
#0  0x00000000004fa710 in selection_changed_at_idle_callback (data=<optimized out>) at caja-icon-container.c:7328
        container = 0xd534330
#1  0x00000035364492a6 in g_main_dispatch (context=0x236e0f0) at gmain.c:3066
        dispatch = 0x3536446150 <g_idle_dispatch>
        was_in_call = 0
        user_data = 0xd534330
        callback = 0x4fa6e0 <selection_changed_at_idle_callback>
        cb_funcs = 0x353672a8e0 <g_source_callback_funcs>
        cb_data = 0x6f97920
        need_destroy = <optimized out>
        current_source_link = {data = 0xbd10590, next = 0x0}
        source = 0xbd10590
        current = 0x233bb30
        i = 0
#2  g_main_context_dispatch (context=context@entry=0x236e0f0) at gmain.c:3642
No locals.
#3  0x0000003536449628 in g_main_context_iterate (context=0x236e0f0, block=block@entry=1, dispatch=dispatch@entry=1, self=<optimized out>) at gmain.c:3713
        max_priority = 200
        timeout = 0
        some_ready = 1
        nfds = <optimized out>
        allocated_nfds = 6
        fds = 0x24c20e0
#4  0x0000003536449a3a in g_main_loop_run (loop=0x24c9230) at gmain.c:3907
        __PRETTY_FUNCTION__ = "g_main_loop_run"
#5  0x0000003543d45c27 in IA__gtk_main () at gtkmain.c:1257
        tmp_list = 0x0
        functions = 0x0
        init = <optimized out>
        loop = 0x24c9230
#6  0x000000000044a8a7 in main (argc=1, argv=0x7fffcc98efa8) at caja-main.c:628
        kill_shell = 0
        no_default_window = 1
        browser_window = 0
        no_desktop = 0
        version = 0
        autostart_mode = <optimized out>
        autostart_id = <optimized out>
        geometry = 0x0
        remaining = 0x0
        perform_self_check = 0
        application = 0x237bc50
        context = <optimized out>
        file = <optimized out>
        fileinfo = <optimized out>
        appinfo = <optimized out>
        uri = <optimized out>
        uris = 0x0
        uris_array = 0x0
        error = 0x0
        i = <optimized out>
        options = {{long_name = 0x55c2e3 "check", short_name = 99 'c', flags = 0, arg = G_OPTION_ARG_NONE, arg_data = 0x7fffcc98ec14, description = 0x55c348 "Perform a quick set of self-check tests.", arg_description = 0x0}, {long_name = 0x57bd75 "version", short_name = 0 '\000', flags = 0, arg = G_OPTION_ARG_NONE, arg_data = 0x7fffcc98ec10, description = 0x55c378 "Show the version of the program.", arg_description = 0x0}, {long_name = 0x560589 "geometry", short_name = 103 'g', flags = 0, arg = G_OPTION_ARG_STRING, arg_data = 0x7fffcc98ec18, description = 0x55c3a0 "Create the initial window with the given geometry.", arg_description = 0x565cb3 "GEOMETRY"}, {long_name = 0x55c225 "no-default-window", short_name = 110 'n', flags = 0, arg = G_OPTION_ARG_NONE, arg_data = 0x7fffcc98ec04, description = 0x55c3d8 "Only create windows for explicitly specified URIs.", arg_description = 0x0}, {long_name = 0x55c237 "no-desktop", short_name = 0 '\000', flags = 0, arg = G_OPTION_ARG_NONE, arg_data = 0x7fffcc98ec0c, description = 0x55c410 "Do not manage the desktop (ignore the preference set in the preferences dialog).", arg_description = 0x0}, {long_name = 0x55d9fb "browser", short_name = 0 '\000', flags = 0, arg = G_OPTION_ARG_NONE, arg_data = 0x7fffcc98ec08, description = 0x55c242 "open a browser window.", arg_description = 0x0}, {long_name = 0x57e585 "quit", short_name = 113 'q', flags = 0, arg = G_OPTION_ARG_NONE, arg_data = 0x7fffcc98ec00, description = 0x55c259 "Quit Caja.", arg_description = 0x0}, {long_name = 0x561b6d "", short_name = 0 '\000', flags = 0, arg = G_OPTION_ARG_STRING_ARRAY, arg_data = 0x7fffcc98ec20, description = 0x0, arg_description = 0x55c264 "[URI...]"}, {long_name = 0x0, short_name = 0 '\000', flags = 0, arg = G_OPTION_ARG_NONE, arg_data = 0x0, description = 0x0, arg_description = 0x0}}
```

A BT of all threads can be found in https://bugzilla.redhat.com/attachment.cgi?id=864278

**Comment by NiceandGently** [comment]
downstream report is closed, because all fedora versions use caja-1.8.x

**Comment by stefano-k** [comment]
Thanks
